ludwig/features/h3_feature.py

```diff
@@ -77,56 +77,30 @@ def add_feature_data(
 
 
 class H3InputFeature(H3BaseFeature, InputFeature):
-    def __init__(self, feature):
-        super().__init__(feature)
+    encoder = 'embed'
 
-        self.encoder = 'embed'
+    def __init__(self, feature, encoder_obj=None):
+        H3BaseFeature.__init__(self, feature)
+        InputFeature.__init__(self)
 
-        encoder_parameters = self.overwrite_defaults(feature)
+        self.overwrite_defaults(feature)
+        if encoder_obj:
+            self.encoder_obj = encoder_obj
+        else:
+            self.encoder_obj = self.initialize_encoder(feature)
 
-        self.encoder_obj = self.get_h3_encoder(encoder_parameters)
 
-    def get_h3_encoder(self, encoder_parameters):
-        return get_from_registry(
-            self.encoder, h3_encoder_registry)(
-            **encoder_parameters
-        )
+    def call(self, inputs, training=None, mask=None):
+        assert isinstance(inputs, tf.Tensor)
+        # assert inputs.dtype == tf.float32 or inputs.dtype == tf.float64
```

Comment:
This should be integers, specifically uint8, but I'm not 100% sure of the casting rules here. I think uint8 can be casted to int32 because of embeddings_lookup needing it for selecting the rows of the embedding matrix, but inputs should arrive here as uint8

Reply:
made the change